web/src/components/LoginModal.tsx

```diff
@@ -0,0 +1,154 @@
+import { useEffect, useRef, useState } from 'react'
+import { setCookie } from '../lib/cookies'
+
+type LoginResponse = {
+  access_token: string
+  refresh_token: string
+}
+
+type Props = {
+  open: boolean
+  onClose: () => void
+  onSuccess: (payload: LoginResponse) => void
+}
+
+const API_BASE = import.meta.env.VITE_API_URL ?? ''
+
+export default function LoginModal({ open, onClose, onSuccess }: Props) {
```

Comment:
looks like it also re-define scroll lock, ESC listener, click-outside logic, and panelRef behavior from modal before. Use DRY, create some base modal and then if you need Login modal expand it, as it looks like it is reusable in your codebase. 
Same rules as in Rails basically

Reply:
something like that:

```JSX
function BaseModal({ open, onClose, title, children }: Props) {
  const panelRef = useRef<HTMLDivElement>(null)

  useModalLock(open)

  useEscapeToClose(open, onClose)

  useClickOutside(panelRef, onClose, open)

  if (!open) return null

  return (
    <div className="overlay" role="dialog" aria-modal="true">
      <div ref={panelRef} className="panel">
        <header className="flex justify-between mb-4">
          <h2>{title}</h2>
          <button onClick={onClose} aria-label="Close">×</button>
        </header>
        {children}
      </div>
    </div>
  )
}
```